bin/plugin/commands/packages.js

```diff
@@ -439,18 +475,22 @@ async function backportCommitsToBranch(
 
 	log( `>> Backporting commits to "${ branchName }".` );
 
-	await git.resetLocalBranchAgainstOrigin(
-		gitWorkingDirectoryPath,
-		branchName
-	);
+	const repo = git( gitWorkingDirectoryPath );
+
+	/*
+	 * Reset any local changes and replace them with the origin branch's copy.
+	 *
+	 * @TODO: Why do we use fetch/checkout/pull here instead of checkout/reset--hard?
+	 *        Is the fetch necessary at all? Do we expect the remote branch to be
+	 *        different now than when we started running this script?
```

Comment:
We've had some issues that I believe were due to a scenario like that, see https://github.com/WordPress/gutenberg/pull/30615. I'd lean towards keeping the fetch; let's maybe update the comment.

Reply:
Ah thank you. That makes sense, especially here where we're back-porting the changes and need to push something fast-forwardable. I'll add a note.